### ISSUE TITLE
Failed to compile: async function *… (v5)

### ISSUE THREAD

**Issue by SMotaal** [jsbin/issues#11]
```apl
main.6eeda05b.js:1 Failed to compile - if this continues, please file a new issue and include this full source and configuration
14:17:57.678 main.6eeda05b.js:1 SyntaxError: unknown: Unexpected token (14:15)
  12 |   
  13 |   
> 14 | 	async function* asynchable() {
     | 	              ^
  15 |     // Prevents infinite loops
  16 |     let limit = 500;
  17 |     
    at t.J.raise (babel.7873188e.chunk.js:1)
    at t.X.unexpected (babel.7873188e.chunk.js:1)
    at t.z.parseFunction (babel.7873188e.chunk.js:1)
    at t.z.parseStatement (babel.7873188e.chunk.js:1)
    at t.z.parseBlockBody (babel.7873188e.chunk.js:1)
    at t.z.parseBlock (babel.7873188e.chunk.js:1)
    at t.z.parseStatement (babel.7873188e.chunk.js:1)
    at t.z.parseBlockBody (babel.7873188e.chunk.js:1)
    at t.z.parseTopLevel (babel.7873188e.chunk.js:1)
    at t.parse (babel.7873188e.chunk.js:1)
```

AsyncGenerators are part of the 2018 specification (download [pdf](https://www.ecma-international.org/publications/files/ECMA-ST/Ecma-262.pdf#sec-async-generator-function-definitions) or [link to draft](https://tc39.github.io/ecma262/#sec-async-generator-function-definitions)) so I really hope we can resolve this issue soon.

👍 Thanks

**Comment by remy** [comment]
Can you share a sample of the source that _should_ work? I've got limited experience with generators so far (I think I've written two generator functions in total!!!).

**Comment by SMotaal** [comment]
Sure... those are all per spec:

Assignment Expression

```js
const a  = async function *a() {};
```

Function Declaration

```js
async function *a() {};
```

Class Method

```js
class A {
  async *a() {}
}
```

Literal Object Method

```js
var o = {
  async *a() {}
};
```

*Arrow notation is no yet in the spec*

**Comment by SMotaal** [comment]
If I simply wrap any of those examples in ``eval(`window.evaled = <properly escaped example>`)`` and followed with `result = window.evaled;` result holds the expected initializer (I tested in v5 in Chrome)

Alternatively (in a new bin) I can simply do this in JS:

```js
eval(`a = async function * generator() {};`) // yuk
console.log({a});
```

**Comment by remy** [comment]
Did you change the language of the JS panel (ie. was it babel or JS)?

**Comment by remy** [comment]
In trying to replicate - I'm getting `Uncaught ReferenceError: regeneratorRuntime is not defined` - which I'm pretty sure means something else is hosed.

Can you share a full screenshot of your devtools console open with the JS panel?

**Comment by SMotaal** [comment]
I made sure it is JS (I intentionally don't want to transpile, I am testing platform support)

<img width="1272" alt="screen shot 2018-07-16 at 4 38 19 pm" src="https://user-images.githubusercontent.com/849613/42782359-aba65cfe-8916-11e8-8381-3506dbf3e1ee.png">

**Comment by SMotaal** [comment]
This is babel+react (but I don't want that anyways):

<img width="1272" alt="screen shot 2018-07-16 at 4 40 33 pm" src="https://user-images.githubusercontent.com/849613/42782458-0554a238-8917-11e8-8f1c-261dbfb354d1.png">

**Comment by SMotaal** [comment]
Here is the same bin using eval wrappers for async generators (JS not babel):

<img width="1272" alt="screen shot 2018-07-16 at 4 41 40 pm" src="https://user-images.githubusercontent.com/849613/42782538-44c2d638-8917-11e8-9a4c-1ecaf20f80fc.png">

**Comment by SMotaal** [comment]
@remy Side question about alpha, so far the only save method that supports revisions in local, in other words, after saving to JS Bin or Gist, saving with the same command from the panel either does nothing or results in a new url, right?

Regarding platforms supporting async generators, I can confirm that Safari is on the list (though it was not a few weeks back)

Here is a gist/bin: https://jsbin.com/gist/7b98512c6381e7838444427e9023a57f?source=javascript

I use gists because I can share them and still be able to edit them on github but I am sure down the road Save to JS Bin is going to be far more useful.

**Comment by remy** [comment]
Took me a while to replicate, but I remembered that the code path is slightly different when you include `import` in the code! Without import, it works fine (though this is obviously not your problem!):

<img width="1280" alt="screen shot 2018-07-17 at 10 10 39" src="https://user-images.githubusercontent.com/13700/42807863-d99c9a40-89a9-11e8-8be9-549f03fd9579.png">

…but _with_ import, it bails because it goes through babel's transform. I'll get into it (and also checkout why babel + react is borking too)

**Comment by SMotaal** [comment]
I was also able to verify that Firefox (at least Beta) also supports async gens.

I honestly applaud your hard work, I always found Babel to bring more question marks for the kind of experimental development I work on all the time... even TypeScript, all those are a bag ❤️'em but hate them. I just never have enough bandwidth for systematically maintaining the moving targets they pose.

>  Without import, it works fine

Just to confirm, I did not use import in my code, just injecting script elements (type≠module)

And to verify, you mean that if my JS include `/^import (.*? from|) ".*?"/` they run through a different pipeline? It would be very reasonable to expect that in order to be able to properly handle dependencies (it is no easy task) 

Would it be possible to come up with a little status indicator in which path is taken (maybe a color coded thing) this way users can know if and when they transition between paths, and they can more actively help in debug offending tooling?

**Comment by SMotaal** [comment]
Any update on this?

**Comment by remy** [comment]
No. Not yet. Not able to get time for it yet. You're welcome to check the
new code out, it's in a much better state than v4!

**Comment by SMotaal** [comment]
I’m happy to find something useful to contribute... let me know if there is something on your mind
